File: Projet_final/JavaProject/src/main/java/model/Salle.java
package model;

import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.persistence.Version;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name = "Salle")
public class Salle {
	@Id
	@SequenceGenerator(name = "seqMateriel", sequenceName = "seq_materiel", initialValue = 101, allocationSize = 1)
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqMateriel")
	@Column(name = "Salle_id")
	@JsonView(JsonViews.Common.class)
	private Long id;
	@Column(name = "Salle_capacite")
	@JsonView(JsonViews.Common.class)
	private Integer capacite;
	@Column(name = "Salle_numero")
	@JsonView(JsonViews.Common.class)
	private String numero;
	@Embedded
	@JsonView(JsonViews.Common.class)
	private Adresse adresse;
	@OneToMany(mappedBy = "salle")
	@JsonView(JsonViews.SalleWithCursus.class)
	private Set<Cursus> cursus;
	@Version
	private int version;

	public Integer getCapacite() {
		return capacite;
	}

	public void setCapacite(Integer capacite) {
		this.capacite = capacite;
	}

	public String getNumero() {
		return numero;
	}

	public void setNumero(String numero) {
		this.numero = numero;
	}

	public Adresse getAdresse() {
		return adresse;
	}

	public void setAdresse(Adresse adresse) {
		this.adresse = adresse;
	}

	public Set<Cursus> getCursus() {
		return cursus;
	}

	public void setCursus(Set<Cursus> cursus) {
		this.cursus = cursus;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}
}

File: Projet_final/angularBT/src/app/model/salle.ts
import { Adresse } from './adresse';
import { Cursus } from './cursus';
export class Salle {
  
  constructor(private _id:string='',
    private _numero:string='',
    private _adresse:Adresse=null,
    private _cursus:Cursus=null,
  private _capacite:string=''){
    
  }
  
  public get id(): string {
    return this._id;
  }
  
  public set id(_id: string) {
    this._id =_id;
  }
  
  public get numero(): string {
    return this._numero;
  }
  
  public set numero(_numero: string) {
    this._numero =_numero;
  }
  
  public get adresse(): Adresse {
    return this._adresse;
  }
  
  public set adresse(_adresse: Adresse) {
    this._adresse =_adresse;
  }
  
  public get capacite(): string {
    return this._capacite;
  }
  
  public set capacite(_capacite: string) {
    this._capacite =_capacite;
  }
  
  public get cursus(): Cursus {
    return this._cursus;
  }
  
  public set cursus(_cursus: Cursus) {
    this._cursus =_cursus;
  }

}